### ISSUE TITLE
pytest failed

### ISSUE THREAD

**Issue by mrakitin** [radiasoft/sirepo#464]
```bash
$ pykern pytest
=========================================================================== test session starts ============================================================================
platform linux2 -- Python 2.7.10, pytest-3.0.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/vagrant/src/radiasoft/sirepo, inifile:
plugins: xdist-1.14, cov-2.3.1
collected 12 items

tests/blank_test.py .
tests/crystal_test.py ...
tests/elegant_import_test.py F
tests/importer_test.py F
tests/server1_test.py F
tests/server_test.py FF
tests/srw_test.py FFF

================================================================================= FAILURES =================================================================================
______________________________________________________________________________ test_importer _______________________________________________________________________________

    def test_importer():
>       from sirepo.template import elegant

tests/elegant_import_test.py:30:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sirepo/template/elegant.py:12: in <module>
    from sirepo import simulation_db
sirepo/simulation_db.py:856: in <module>
    _init()
sirepo/simulation_db.py:720: in _init
    nfs_sleep=(0.5, float, 'Seconds sleep per hack_nfs_write_status poll'),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kwargs = {'nfs_sleep': (0.5, <type 'float'>, 'Seconds sleep per hack_nfs_write_status poll'), 'nfs_tries': (10, <type 'int'>, 'How many times to poll in hack_nfs_write_status')}
m = <module 'sirepo.simulation_db' from '/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.pyc'>

    def init(**kwargs):
        """Declares and initializes config params for calling module.

        Args:
            kwargs (dict): param name to (default, parser, docstring)

        Returns:
            Params: `pkcollections.OrderedMapping` populated with param values
        """
        if '_caller_module' in kwargs:
            # Internal use only: _values() calls init() to initialize pkconfig.cfg
            m = kwargs['_caller_module']
            del kwargs['_caller_module']
        else:
            if pkinspect.is_caller_main():
                print(
                    'pkconfig.init() called from __main__; cannot configure, ignoring',
                    file=sys.stderr)
                return None
            m = pkinspect.caller_module()
        assert pkinspect.root_package(m) in _load_path, \
>           '{}: module root not in load_path ({})'.format(m.__name__, _load_path)
E           AssertionError: sirepo.simulation_db: module root not in load_path (['pykern'])

../pykern/pykern/pkconfig.py:350: AssertionError
______________________________________________________________________________ test_importer _______________________________________________________________________________

    def test_importer():
        from sirepo.importer import import_python
        dat_dir = py.path.local(pkresource.filename('static/dat/', import_python))
        with pkunit.save_chdir_work():
            work_dir = py.path.local('.')
            for f in glob.glob(str(dat_dir.join('mirror_*d.dat'))):
                py.path.local(f).copy(work_dir)
            for b in sorted(_TESTS.keys()):
                base_py = '{}.py'.format(_TESTS[b][0])
                code = pkio.read_text(pkunit.data_dir().join(base_py))
                error, actual = import_python(
                    code,
                    tmp_dir=str(work_dir),
                    lib_dir=str(work_dir),
                    user_filename=r'c:\anything\{}.anysuffix'.format(_TESTS[b][0]),
                    arguments=_TESTS[b][1],
                )
                assert not error, \
                    '{}: should import with an error: {}'.format(base_py, error)
                actual['version'] = 'IGNORE-VALUE'
                assert not error, \
                    '{}: should be valid input'.format(base_py)
>               pkunit.assert_object_with_json(b, actual)

tests/importer_test.py:62:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

basename = 'amx'
actual = '{\n    "models": {\n        "beamline": [\n            {\n                "horizontalOffset": 0,\n                "ho...           "verticalSymmetry": -1\n        }\n    },\n    "simulationType": "srw",\n    "version": "IGNORE-VALUE"\n}\n'

    def assert_object_with_json(basename, actual):
        """Converts actual to JSON and compares with data_dir/basename.json

        Reads data_dir/basename.json and compares with actual
        converted to json. Trailing newline is managed properly. The
        keys are sorted and indentation is 4. actual written to work_dir.

        Args:
            expected_basename (str): file to be found in data_dir with json suffix
            actual (object): to be serialized as json
        """
        actual = pkdpretty(actual)
        fn = '{}.json'.format(basename)
        pkio.write_text(work_dir().join(fn), actual)
        expect = pkio.read_text(data_dir().join(fn))
>       assert expect == actual
E       AssertionError

../pykern/pykern/pkunit.py:53: AssertionError
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
[0.9999999999999999, -2.5866939514398794e-09, -1.6653345369377348e-16]
[2.5866939514398827e-09, 1.0, -1.1486052770794924e-08]
[1.6653345369377348e-16, 1.1486052770794924e-08, 0.9999999999999999]
___________________________________________________________________________ test_1_serial_stomp ____________________________________________________________________________

    def test_1_serial_stomp():
        from pykern.pkdebug import pkdp, pkdpretty
        from pykern.pkunit import pkfail, pkok
>       from sirepo import sr_unit

tests/server1_test.py:18:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sirepo/sr_unit.py:10: in <module>
    from sirepo import server
sirepo/server.py:13: in <module>
    from sirepo import simulation_db
sirepo/simulation_db.py:856: in <module>
    _init()
sirepo/simulation_db.py:720: in _init
    nfs_sleep=(0.5, float, 'Seconds sleep per hack_nfs_write_status poll'),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kwargs = {'nfs_sleep': (0.5, <type 'float'>, 'Seconds sleep per hack_nfs_write_status poll'), 'nfs_tries': (10, <type 'int'>, 'How many times to poll in hack_nfs_write_status')}
m = <module 'sirepo.simulation_db' from '/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.pyc'>

    def init(**kwargs):
        """Declares and initializes config params for calling module.

        Args:
            kwargs (dict): param name to (default, parser, docstring)

        Returns:
            Params: `pkcollections.OrderedMapping` populated with param values
        """
        if '_caller_module' in kwargs:
            # Internal use only: _values() calls init() to initialize pkconfig.cfg
            m = kwargs['_caller_module']
            del kwargs['_caller_module']
        else:
            if pkinspect.is_caller_main():
                print(
                    'pkconfig.init() called from __main__; cannot configure, ignoring',
                    file=sys.stderr)
                return None
            m = pkinspect.caller_module()
        assert pkinspect.root_package(m) in _load_path, \
>           '{}: module root not in load_path ({})'.format(m.__name__, _load_path)
E           AssertionError: sirepo.simulation_db: module root not in load_path (['pykern'])

../pykern/pykern/pkconfig.py:350: AssertionError
________________________________________________________________________________ test_basic ________________________________________________________________________________

    def test_basic():
>       from sirepo import sr_unit

tests/server_test.py:12:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sirepo/sr_unit.py:10: in <module>
    from sirepo import server
sirepo/server.py:13: in <module>
    from sirepo import simulation_db
sirepo/simulation_db.py:856: in <module>
    _init()
sirepo/simulation_db.py:720: in _init
    nfs_sleep=(0.5, float, 'Seconds sleep per hack_nfs_write_status poll'),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kwargs = {'nfs_sleep': (0.5, <type 'float'>, 'Seconds sleep per hack_nfs_write_status poll'), 'nfs_tries': (10, <type 'int'>, 'How many times to poll in hack_nfs_write_status')}
m = <module 'sirepo.simulation_db' from '/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.pyc'>

    def init(**kwargs):
        """Declares and initializes config params for calling module.

        Args:
            kwargs (dict): param name to (default, parser, docstring)

        Returns:
            Params: `pkcollections.OrderedMapping` populated with param values
        """
        if '_caller_module' in kwargs:
            # Internal use only: _values() calls init() to initialize pkconfig.cfg
            m = kwargs['_caller_module']
            del kwargs['_caller_module']
        else:
            if pkinspect.is_caller_main():
                print(
                    'pkconfig.init() called from __main__; cannot configure, ignoring',
                    file=sys.stderr)
                return None
            m = pkinspect.caller_module()
        assert pkinspect.root_package(m) in _load_path, \
>           '{}: module root not in load_path ({})'.format(m.__name__, _load_path)
E           AssertionError: sirepo.simulation_db: module root not in load_path (['pykern'])

../pykern/pykern/pkconfig.py:350: AssertionError
_________________________________________________________________________________ test_srw _________________________________________________________________________________

    def test_srw():
        from pykern import pkio
        from pykern.pkdebug import pkdpretty
>       from sirepo import sr_unit

tests/server_test.py:23:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sirepo/sr_unit.py:10: in <module>
    from sirepo import server
sirepo/server.py:13: in <module>
    from sirepo import simulation_db
sirepo/simulation_db.py:856: in <module>
    _init()
sirepo/simulation_db.py:720: in _init
    nfs_sleep=(0.5, float, 'Seconds sleep per hack_nfs_write_status poll'),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kwargs = {'nfs_sleep': (0.5, <type 'float'>, 'Seconds sleep per hack_nfs_write_status poll'), 'nfs_tries': (10, <type 'int'>, 'How many times to poll in hack_nfs_write_status')}
m = <module 'sirepo.simulation_db' from '/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.pyc'>

    def init(**kwargs):
        """Declares and initializes config params for calling module.

        Args:
            kwargs (dict): param name to (default, parser, docstring)

        Returns:
            Params: `pkcollections.OrderedMapping` populated with param values
        """
        if '_caller_module' in kwargs:
            # Internal use only: _values() calls init() to initialize pkconfig.cfg
            m = kwargs['_caller_module']
            del kwargs['_caller_module']
        else:
            if pkinspect.is_caller_main():
                print(
                    'pkconfig.init() called from __main__; cannot configure, ignoring',
                    file=sys.stderr)
                return None
            m = pkinspect.caller_module()
        assert pkinspect.root_package(m) in _load_path, \
>           '{}: module root not in load_path ({})'.format(m.__name__, _load_path)
E           AssertionError: sirepo.simulation_db: module root not in load_path (['pykern'])

../pykern/pykern/pkconfig.py:350: AssertionError
____________________________________________________________________ test_find_height_profile_dimension ____________________________________________________________________

    def test_find_height_profile_dimension():
>       from sirepo.template import srw

tests/srw_test.py:17:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sirepo/template/srw.py:14: in <module>
    from sirepo import simulation_db
sirepo/simulation_db.py:856: in <module>
    _init()
sirepo/simulation_db.py:720: in _init
    nfs_sleep=(0.5, float, 'Seconds sleep per hack_nfs_write_status poll'),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kwargs = {'nfs_sleep': (0.5, <type 'float'>, 'Seconds sleep per hack_nfs_write_status poll'), 'nfs_tries': (10, <type 'int'>, 'How many times to poll in hack_nfs_write_status')}
m = <module 'sirepo.simulation_db' from '/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.pyc'>

    def init(**kwargs):
        """Declares and initializes config params for calling module.

        Args:
            kwargs (dict): param name to (default, parser, docstring)

        Returns:
            Params: `pkcollections.OrderedMapping` populated with param values
        """
        if '_caller_module' in kwargs:
            # Internal use only: _values() calls init() to initialize pkconfig.cfg
            m = kwargs['_caller_module']
            del kwargs['_caller_module']
        else:
            if pkinspect.is_caller_main():
                print(
                    'pkconfig.init() called from __main__; cannot configure, ignoring',
                    file=sys.stderr)
                return None
            m = pkinspect.caller_module()
        assert pkinspect.root_package(m) in _load_path, \
>           '{}: module root not in load_path ({})'.format(m.__name__, _load_path)
E           AssertionError: sirepo.simulation_db: module root not in load_path (['pykern'])

../pykern/pykern/pkconfig.py:350: AssertionError
______________________________________________________________________ test_find_tab_undulator_length ______________________________________________________________________

    def test_find_tab_undulator_length():
>       from sirepo.template import srw

tests/srw_test.py:25:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sirepo/template/srw.py:14: in <module>
    from sirepo import simulation_db
sirepo/simulation_db.py:856: in <module>
    _init()
sirepo/simulation_db.py:720: in _init
    nfs_sleep=(0.5, float, 'Seconds sleep per hack_nfs_write_status poll'),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kwargs = {'nfs_sleep': (0.5, <type 'float'>, 'Seconds sleep per hack_nfs_write_status poll'), 'nfs_tries': (10, <type 'int'>, 'How many times to poll in hack_nfs_write_status')}
m = <module 'sirepo.simulation_db' from '/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.pyc'>

    def init(**kwargs):
        """Declares and initializes config params for calling module.

        Args:
            kwargs (dict): param name to (default, parser, docstring)

        Returns:
            Params: `pkcollections.OrderedMapping` populated with param values
        """
        if '_caller_module' in kwargs:
            # Internal use only: _values() calls init() to initialize pkconfig.cfg
            m = kwargs['_caller_module']
            del kwargs['_caller_module']
        else:
            if pkinspect.is_caller_main():
                print(
                    'pkconfig.init() called from __main__; cannot configure, ignoring',
                    file=sys.stderr)
                return None
            m = pkinspect.caller_module()
        assert pkinspect.root_package(m) in _load_path, \
>           '{}: module root not in load_path ({})'.format(m.__name__, _load_path)
E           AssertionError: sirepo.simulation_db: module root not in load_path (['pykern'])

../pykern/pykern/pkconfig.py:350: AssertionError
__________________________________________________________________________ test_prepare_aux_files __________________________________________________________________________

    def test_prepare_aux_files():
>       from sirepo.template import srw

tests/srw_test.py:39:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sirepo/template/srw.py:14: in <module>
    from sirepo import simulation_db
sirepo/simulation_db.py:856: in <module>
    _init()
sirepo/simulation_db.py:720: in _init
    nfs_sleep=(0.5, float, 'Seconds sleep per hack_nfs_write_status poll'),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kwargs = {'nfs_sleep': (0.5, <type 'float'>, 'Seconds sleep per hack_nfs_write_status poll'), 'nfs_tries': (10, <type 'int'>, 'How many times to poll in hack_nfs_write_status')}
m = <module 'sirepo.simulation_db' from '/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.pyc'>

    def init(**kwargs):
        """Declares and initializes config params for calling module.

        Args:
            kwargs (dict): param name to (default, parser, docstring)

        Returns:
            Params: `pkcollections.OrderedMapping` populated with param values
        """
        if '_caller_module' in kwargs:
            # Internal use only: _values() calls init() to initialize pkconfig.cfg
            m = kwargs['_caller_module']
            del kwargs['_caller_module']
        else:
            if pkinspect.is_caller_main():
                print(
                    'pkconfig.init() called from __main__; cannot configure, ignoring',
                    file=sys.stderr)
                return None
            m = pkinspect.caller_module()
        assert pkinspect.root_package(m) in _load_path, \
>           '{}: module root not in load_path ({})'.format(m.__name__, _load_path)
E           AssertionError: sirepo.simulation_db: module root not in load_path (['pykern'])

../pykern/pykern/pkconfig.py:350: AssertionError
========================================================================== pytest-warning summary ==========================================================================
WC1 /home/vagrant/src/radiasoft/sirepo/tests/elegant_import_test.py cannot collect test class 'TestFlaskRequest' because it has a __init__ constructor
========================================================== 8 failed, 4 passed, 1 pytest-warnings in 6.37 seconds ===========================================================
```

**Comment by robnagler** [comment]
Do you have the latest pykern?

**Comment by mrakitin** [comment]
Yes.

**Comment by robnagler** [comment]
What happens when you run `python setup.py test`?

**Comment by mrakitin** [comment]
There were a couple of errors regarding disabled rabbitmq and celery, after I enabled them, there are the following errors:
```python
16:06 $ python setup.py test
running test
running egg_info
writing requirements to sirepo.egg-info/requires.txt
writing sirepo.egg-info/PKG-INFO
writing top-level names to sirepo.egg-info/top_level.txt
writing dependency_links to sirepo.egg-info/dependency_links.txt
writing entry points to sirepo.egg-info/entry_points.txt
reading manifest file 'sirepo.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'sirepo.egg-info/SOURCES.txt'
running build_ext
=========================================================================== test session starts ============================================================================
platform linux2 -- Python 2.7.10, pytest-3.0.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/vagrant/src/radiasoft/sirepo, inifile:
plugins: xdist-1.14, cov-2.3.1
collected 12 items

tests/blank_test.py .
tests/crystal_test.py ...
tests/elegant_import_test.py .
tests/importer_test.py F
tests/server1_test.py F
tests/server_test.py FF
tests/srw_test.py ...

================================================================================= FAILURES =================================================================================
______________________________________________________________________________ test_importer _______________________________________________________________________________

    def test_importer():
        from sirepo.importer import import_python
        dat_dir = py.path.local(pkresource.filename('static/dat/', import_python))
        with pkunit.save_chdir_work():
            work_dir = py.path.local('.')
            for f in glob.glob(str(dat_dir.join('mirror_*d.dat'))):
                py.path.local(f).copy(work_dir)
            for b in sorted(_TESTS.keys()):
                base_py = '{}.py'.format(_TESTS[b][0])
                code = pkio.read_text(pkunit.data_dir().join(base_py))
                error, actual = import_python(
                    code,
                    tmp_dir=str(work_dir),
                    lib_dir=str(work_dir),
                    user_filename=r'c:\anything\{}.anysuffix'.format(_TESTS[b][0]),
                    arguments=_TESTS[b][1],
                )
                assert not error, \
                    '{}: should import with an error: {}'.format(base_py, error)
                actual['version'] = 'IGNORE-VALUE'
                assert not error, \
                    '{}: should be valid input'.format(base_py)
>               pkunit.assert_object_with_json(b, actual)

tests/importer_test.py:62:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

basename = 'amx'
actual = '{\n    "models": {\n        "beamline": [\n            {\n                "horizontalOffset": 0,\n                "ho...           "verticalSymmetry": -1\n        }\n    },\n    "simulationType": "srw",\n    "version": "IGNORE-VALUE"\n}\n'

    def assert_object_with_json(basename, actual):
        """Converts actual to JSON and compares with data_dir/basename.json

        Reads data_dir/basename.json and compares with actual
        converted to json. Trailing newline is managed properly. The
        keys are sorted and indentation is 4. actual written to work_dir.

        Args:
            expected_basename (str): file to be found in data_dir with json suffix
            actual (object): to be serialized as json
        """
        actual = pkdpretty(actual)
        fn = '{}.json'.format(basename)
        pkio.write_text(work_dir().join(fn), actual)
        expect = pkio.read_text(data_dir().join(fn))
>       assert expect == actual
E       AssertionError

../pykern/pykern/pkunit.py:53: AssertionError
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
[0.9999999999999999, -2.5866939514398794e-09, -1.6653345369377348e-16]
[2.5866939514398827e-09, 1.0, -1.1486052770794924e-08]
[1.6653345369377348e-16, 1.1486052770794924e-08, 0.9999999999999999]
--------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------
Oct 03 17:20:36 13816     0 ../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:805:_new_conn INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): raw.githubusercontent.com
Oct 03 17:20:36 13816     0 ../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:805:_new_conn INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): raw.githubusercontent.com
Oct 03 17:20:36 13816     0 ../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:805:_new_conn INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): raw.githubusercontent.com
Oct 03 17:20:36 13816     0 ../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:805:_new_conn INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): raw.githubusercontent.com
___________________________________________________________________________ test_1_serial_stomp ____________________________________________________________________________

    def test_1_serial_stomp():
        from pykern.pkdebug import pkdp, pkdpretty
        from pykern.pkunit import pkfail, pkok
        from sirepo import sr_unit
        import copy

>       fc = sr_unit.flask_client()

tests/server1_test.py:21:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sirepo/sr_unit.py:30: in flask_client
    with pkio.save_chdir(pkunit.work_dir()):
../pykern/pykern/pkunit.py:242: in work_dir
    return _base_dir(_WORK_DIR_SUFFIX).ensure(dir=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

postfix = '_work'

    def _base_dir(postfix):
        """Base name with directory.

        Args:
            postfix (str): what to append to base (``_data`` or ``_work``).

        Returns:
            py.path.local: base directory with postfix
        """
        m = module_under_test or pkinspect.caller_module()
        filename = py.path.local(m.__file__)
        b = re.sub(r'_test$|^test_', '', filename.purebasename)
        assert b != filename.purebasename, \
>           '{}: module name must end in _test'.format(filename)
E       AssertionError: /home/vagrant/src/radiasoft/sirepo/sirepo/sr_unit.pyc: module name must end in _test

../pykern/pykern/pkunit.py:258: AssertionError
________________________________________________________________________________ test_basic ________________________________________________________________________________

    def test_basic():
        from sirepo import sr_unit
>       fc = sr_unit.flask_client()

tests/server_test.py:13:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sirepo/sr_unit.py:30: in flask_client
    with pkio.save_chdir(pkunit.work_dir()):
../pykern/pykern/pkunit.py:242: in work_dir
    return _base_dir(_WORK_DIR_SUFFIX).ensure(dir=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

postfix = '_work'

    def _base_dir(postfix):
        """Base name with directory.

        Args:
            postfix (str): what to append to base (``_data`` or ``_work``).

        Returns:
            py.path.local: base directory with postfix
        """
        m = module_under_test or pkinspect.caller_module()
        filename = py.path.local(m.__file__)
        b = re.sub(r'_test$|^test_', '', filename.purebasename)
        assert b != filename.purebasename, \
>           '{}: module name must end in _test'.format(filename)
E       AssertionError: /home/vagrant/src/radiasoft/sirepo/sirepo/sr_unit.pyc: module name must end in _test

../pykern/pykern/pkunit.py:258: AssertionError
_________________________________________________________________________________ test_srw _________________________________________________________________________________

    def test_srw():
        from pykern import pkio
        from pykern.pkdebug import pkdpretty
        from sirepo import sr_unit
        import json

>       fc = sr_unit.flask_client()

tests/server_test.py:26:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sirepo/sr_unit.py:30: in flask_client
    with pkio.save_chdir(pkunit.work_dir()):
../pykern/pykern/pkunit.py:242: in work_dir
    return _base_dir(_WORK_DIR_SUFFIX).ensure(dir=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

postfix = '_work'

    def _base_dir(postfix):
        """Base name with directory.

        Args:
            postfix (str): what to append to base (``_data`` or ``_work``).

        Returns:
            py.path.local: base directory with postfix
        """
        m = module_under_test or pkinspect.caller_module()
        filename = py.path.local(m.__file__)
        b = re.sub(r'_test$|^test_', '', filename.purebasename)
        assert b != filename.purebasename, \
>           '{}: module name must end in _test'.format(filename)
E       AssertionError: /home/vagrant/src/radiasoft/sirepo/sirepo/sr_unit.pyc: module name must end in _test

../pykern/pykern/pkunit.py:258: AssertionError
========================================================================== pytest-warning summary ==========================================================================
WC1 /home/vagrant/src/radiasoft/sirepo/tests/elegant_import_test.py cannot collect test class 'TestFlaskRequest' because it has a __init__ constructor
========================================================== 4 failed, 8 passed, 1 pytest-warnings in 10.49 seconds ==========================================================
```

**Comment by robnagler** [comment]
Maybe you need to:

```python
cd ~/src/radiasoft/pykern
pip install -e .
```

pykern now implements a pytest plugin so you may need the pip install.

**Comment by mrakitin** [comment]
Yes, I did that. Yesterday I fixed the SRW importer tests (#471), but there is still one issue with the server test:
```py
14:32 $ python setup.py test
running test
running egg_info
writing requirements to sirepo.egg-info/requires.txt
writing sirepo.egg-info/PKG-INFO
writing top-level names to sirepo.egg-info/top_level.txt
writing dependency_links to sirepo.egg-info/dependency_links.txt
writing entry points to sirepo.egg-info/entry_points.txt
reading manifest file 'sirepo.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'sirepo.egg-info/SOURCES.txt'
running build_ext
============================================================================ test session starts ============================================================================
platform linux2 -- Python 2.7.10, pytest-3.0.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/vagrant/src/radiasoft/sirepo, inifile:
plugins: xdist-1.14, cov-2.3.1, pykern-20160928.212331
collected 12 items

tests/blank_test.py .
tests/crystal_test.py ...
tests/elegant_import_test.py .
tests/importer_test.py .
tests/server1_test.py .
tests/server_test.py F.
tests/srw_test.py ...

================================================================================= FAILURES ==================================================================================
________________________________________________________________________________ test_basic _________________________________________________________________________________
def test_basic():
        from sirepo import sr_unit
        fc = sr_unit.flask_client()

        resp = fc.get('/')
>       assert resp.status_code == 404, \
            'There should not be a / route'
E       AssertionError: There should not be a / route
E       assert 200 == 404
E        +  where 200 = <Response streamed [200 OK]>.status_code

tests/server_test.py:16: AssertionError
========================================================================== pytest-warning summary ===========================================================================
WC1 /home/vagrant/src/radiasoft/sirepo/tests/elegant_import_test.py cannot collect test class 'TestFlaskRequest' because it has a __init__ constructor
========================================================== 1 failed, 11 passed, 1 pytest-warnings in 88.82 seconds ==========================================================
```

**Comment by robnagler** [comment]
Fixed the test in 4a7119b. I don't think pykern pytest works any more, because there are weird things with setting up pytest (see below). python setup.py seems to do the right thing. 

You can run py.test on individual files. Running the collector breaks. I'll add a new bug

**Comment by mrakitin** [comment]
It seems to be fixed in https://github.com/radiasoft/sirepo/commit/87bb55ef0803db04600f1cd4651fbe31fb546068. Thanks @robnagler.
